config/routes.rb

```diff
@@ -9,6 +9,7 @@
     # On nest ces routes car on a besoin de group_id pour new, create, edit et update
     resources :courses, only: [:new, :create, :edit, :update]
     patch "courses/:id/publish", to: "courses#publish", as: :course_publish
```

Comment:
Je pense que tu n'as pas besoin de nester tes deux routes `courses/:id/publish` et `courses/:id/depublish` sous group && course car à aucun moment tu utilises l'id du group dans les actions `publish` et `depublish` dans le contrôleur `groups`. 

Par conséquent, je pense que tu peux simplifier tes routes de la façon suivante (à écrire juste après `root to: 'pages#home'` par exemple):

`resources :courses do`
     `member do`
        `get "publish", to: "courses#publish", as: :course_publish`
        `get "depublish", to: "courses#depublish", as: :course_depublish`
     `end`
`end`

=> cela va générer les routes suivantes `courses/:id/publish` et `courses/:id/depublish`

=> impact direct dans ton contrôleur de groups: 

      1. tu peux sortir `:publish` du `before_action :find_group`
      2. tu peux ajouter `:depublish` dans le `before_action :find_course`

=> impact direct dans ta vue de course: 

les liens `group_course_publish_path(@group, @course)` et `group_course_depublish_path(@group, @course)` deviennent respectivement: `course_publish_path(@course)` et `course_depublish_path(@group, @course)`.